mayastor/src/bdev/nexus/nexus_child.rs

```diff
@@ -266,10 +272,18 @@ impl NexusChild {
         }
         self.open(parent_size).map(|s| {
             self.status_reasons.offline(false);
+            NexusChild::save_state_change();
             s
         })
     }
 
+    /// Save the state of the children to the config file
+    pub(crate) fn save_state_change() {
+        if ChildStatusConfig::save().is_err() {
+            error!("Failed to save child status information");
```

Comment:
should this be a fatal error?

Reply:
See response above about handling error conditions. (Defer the work to a future work item)